cmd/healthcheck-handler.go

```diff
@@ -51,8 +54,29 @@ func LivenessCheckHandler(w http.ResponseWriter, r *http.Request) {
 		writeResponse(w, http.StatusServiceUnavailable, nil, mimeNone)
 		return
 	}
-	// List buckets is unsuccessful, means server is having issues, send 503 service unavailable
-	if _, err := objLayer.ListBuckets(context.Background()); err != nil {
+	var totalLocalDisks int
+	for _, endpoint := range globalEndpoints {
+		if endpoint.IsLocal {
+			totalLocalDisks++
+		}
+	}
+	var erroredDisks int
+	for _, endpoint := range globalEndpoints {
+		// Check only if local disks are accessible, we do not have
+		// to reach to rest of the other servers in a distributed setup.
+		if endpoint.IsLocal {
+			// Simply attempt a stat to backend, any error resulting
+			// from this Stat() operation is considered as backend
+			// is not available, count them as errors.
+			if _, err := os.Stat(endpoint.Path); err != nil {
+				logger.LogIf(context.Background(), err)
```

Comment:
Instead of `context.Background()`, we can create a `context` and pass it here, it will have more information like `request-id` and the actual `API` (It can also be obtained from the stacktrace)